sentry/src/main/java/io/sentry/IScopes.java

```diff
@@ -326,6 +333,19 @@ default void addBreadcrumb(@NotNull String message, @NotNull String category) {
    */
   void withScope(@NotNull ScopeCallback callback);
 
+  /**
+   * Runs the callback with a new isolation scope which gets dropped at the end. Current scope is
+   * also forked.
+   *
+   * <p>If you're using the Sentry SDK in globalHubMode (defaults to true on Android) {@link
+   * Sentry#init(Sentry.OptionsConfiguration, boolean)} calling withScope is discouraged, as scope
+   * changes may be dropped when executed in parallel. Use {@link IScopes#configureScope(ScopeType,
+   * ScopeCallback)} instead.
+   *
+   * @param callback the callback
+   */
+  void withIsolationScope(@NotNull ScopeCallback callback);
```

Comment:
This is new. Having a `ScopeType` param for `withScope` could be more confusing.